src/lib.rs

```diff
@@ -91,6 +91,8 @@
 
 #![doc(html_root_url = "https://docs.rs/tokio-core/0.1")]
 #![deny(missing_docs)]
+#![deny(warnings)]
+#![cfg_attr(test, allow(deprecated))]
```

Comment:
I'm not totally sure why this line is necessary-- without it, `cargo test` fails with "use of deprecated item: moved to the `tokio-io` crate", but it doesn't point to any actual code.